src/problem1.rb

```diff
@@ -0,0 +1,7 @@
+class Problem1
+  def even_numbers_sum(array)
+    return 0 if array.nil?
+    sum = array.select(&:even?).reduce(:+)
+    sum == nil ? 0 : sum
```

Comment:
This can be furthure simplified. Check @alexandru-calinoiu's comment. Also no need to guard for nil. Problems state that you are given an array of size N